crates/primitives/src/db.rs

```diff
@@ -1,27 +1,30 @@
-pub mod components;
-
 use crate::AccountInfo;
 use crate::U256;
 use crate::{Account, Bytecode};
 use crate::{B160, B256};
 use auto_impl::auto_impl;
 use hashbrown::HashMap as Map;
 
+pub mod components;
```

Comment:
Why this style and not moving `pub used componnets` behind `pub mod`?

Reply:
I don't think there's a style guide for this, but I like when it's `mod x; use x::*;` groups, like: <https://github.com/alloy-rs/core/blob/d9d0ded3ca4e2dde992dad1afe6410015158d6b1/crates/syn-solidity/src/lib.rs>

Reply:
I would prefer a little bit of a different order.

```
// use if needed. or maybe even put reexports.
use std::..

// list all modules in one place
mod components;
mod tests;

// reexport module items here.
pub use components::*
```

Something like this is easier to my eyes: https://github.com/bluealloy/revm/blob/main/crates/revm/src/db/states.rs
having `mod *` grouped together.

Just bikesheding, this change in general is nice